src/bin/pg_autoctl/cli_enable_disable.c

```diff
@@ -402,3 +432,566 @@ cli_disable_maintenance(int argc, char **argv)
+	if (sslCommandLineOptions == SSL_CLI_UNKNOWN)
+	{
+		log_fatal("Explicit SSL choice is required: please use either "
+				  "--ssl-self-signed or provide your certificates "
+				  "using --ssl-ca-file, --ssl-crl-file, "
+				  "--server-key, and --server-crt (or use --no-ssl if you "
+				  "are very sure that you do not want encrypted traffic)");
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!pgsetup_validate_ssl_settings(&(options.pgSetup)))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	/* publish our option parsing in the global variable */
+	keeperOptions = options;
+
+	return optind;
+}
+
+
+/*
+ * cli_enable_ssl enables SSL setup on this node.
+ *
+ *  - edit our Postgres configuration with the given SSL files and options
+ *  - when run on a keeper, checks that the monitor accepts ssl connections
+ *  - edits our configuration at pg_autoctl.conf
+ */
+static void
+cli_enable_ssl(int argc, char **argv)
+{
+	KeeperConfig options = keeperOptions;
+
+	bool missingPgdataIsOk = true;
+	bool pgIsNotRunningIsOk = true;
+	bool monitorDisabledIsOk = true;
+
+	switch (ProbeConfigurationFileRole(options.pathnames.config))
+	{
+		case PG_AUTOCTL_ROLE_MONITOR:
+		{
+			MonitorConfig mconfig = { 0 };
+			PostgresSetup *pgSetup = &(mconfig.pgSetup);
+			LocalPostgresServer postgres = { 0 };
+
+			bool reloadedService = false;
+
+			if (!monitor_config_init_from_pgsetup(&mconfig,
+												  &options.pgSetup,
+												  missingPgdataIsOk,
+												  pgIsNotRunningIsOk))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* now override current on-file settings with CLI ssl options */
+			pgSetup->ssl = options.pgSetup.ssl;
+
+			local_postgres_init(&postgres, pgSetup);
+
+			/* update the Postgres SSL setup and maybe create the certificate */
+			if (!update_ssl_configuration(&postgres, mconfig.nodename))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_INTERNAL_ERROR);
+			}
+
+			/* make sure that the new SSL files are part of the setup */
+			mconfig.pgSetup.ssl = postgres.postgresSetup.ssl;
+
+			/* update the monitor's configuration to use SSL */
+			if (!monitor_config_write_file(&mconfig))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			if (file_exists(mconfig.pathnames.pid))
+			{
+				reloadedService = cli_pg_autoctl_reload(mconfig.pathnames.pid);
+			}
+
+			/* display a nice summary to our users */
+			log_info("Successfully enabled new SSL configuration:");
+			log_info("  SSL is now %s",
+					 pgSetup->ssl.active ? "active" : "disabled");
+
+			if (pgSetup->ssl.createSelfSignedCert)
+			{
+				log_info("  Self-Signed certificates have been created and "
+						 "deployed in Postgres configuration settings "
+						 "ssl_key_file and ssl_cert_file");
+			}
+
+			if (reloadedService)
+			{
+				log_info("  pg_autoctl service has been signaled to reload "
+						 "its configuration");
+			}
```

Comment:
Done.

Reply:
The change you did is not what I meant. I think it's good too and we should keep it. But it doesn't cover one important case, the case where the service is not running at all. Especially on the monitor this seems like a situation that could occur, since it does not do much there.

What I meant is add an error line to the summary. That if it was not reloaded the service should be (re)started, to apply the changes.

Reply:
Done now.